pshtt/pshtt.py

```diff
@@ -344,6 +343,14 @@ def basic_check(endpoint):
                 (subdomain_original != subdomain_immediate)
             )
 
+            # We're interested in whether an endpoint redirects to the www version
+            # of itself (not whether it redirects to www prepended to any other
+            # hostname, even within the same parent domain).
+            endpoint.redirect_immediately_to_www = (
+                subdomain_immediate.startswith("www.") and
+                (re.sub("www\.", "", subdomain_immediate) == subdomain_original)
```

Comment:
Would just `subdomain_intermediate == 'www.'+subdomain_original` be easier and more efficient here?

Reply:
Yeah, probably! :) Went ahead and made this improvement (using interpolation) in #155.